util/eventbus/subscribe.go

```diff
@@ -134,7 +134,8 @@ func (s *subscribeState) subscribeTypes() []reflect.Type {
 	return ret
 }
 
-func (s *subscribeState) addSubscriber(t reflect.Type, sub subscriber) {
+func (s *subscribeState) addSubscriber(sub subscriber) {
+	t := sub.subscribeType()
```

Comment:
We might as well put this inside the lock, these events are infrequent enough we don't need to shave cycles.